TripCheckins/Application flow/AppCoordinator.swift

```diff
@@ -41,10 +37,20 @@ class AppCoordinator {
     
     // MARK: - Private
     private func showAuthorizationViewController() {
+        guard let authorizationFormPresenter = self.navigationController.viewControllers.last else {
```

Comment:
So if I'm understanding things correctly, if showAuthorizationViewController is called without showTrip being called before it, the user will be stuck. 

I will try to consider a more explicit approach than the authorizationPendingAction closure. Wouldn't this be simpler:
`private func showAuthorizationViewController(tripID: String)
` ?

Reply:
Depends what you mean by "the user will be stuck". This private method just triggers the authorization flow and saves the token if it is available during that process, then the user can continue to interact with the app. I haven't considered future usages because as the description implies it is likely to change soon. But I agree that it can be more explicit, what about passing the closure to the method instead of using a property?

Reply:
passing the closure to the method would be much better in my opinion